src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/OverlappingTrainingDateRequestController.cs

```diff
@@ -47,13 +47,59 @@ IFeatureTogglesService<ProviderFeatureToggle> featureTogglesService
             _featureTogglesService = featureTogglesService;
         }
 
+        [HttpGet]
+        [Route("overlap-options-with-pending-request")]
+        public IActionResult DraftApprenticeshipOverlapOptionsWithPendingRequest(DraftApprenticeshipOverlapOptionWithPendingRequest request)
+        {
+            var featureToggleEnabled = _featureTogglesService.GetFeatureToggle(ProviderFeature.OverlappingTrainingDateWithoutPrefix).IsEnabled;
+            var vm = new DraftApprenticeshipOverlapOptionWithPendingRequestViewModel()
+            {
+                CohortReference = request.CohortReference,
+                DraftApprenticeshipHashedId = request.DraftApprenticeshipHashedId,
+                DraftApprenticeshipId = request.DraftApprenticeshipId,
+                OverlappingTrainingDateRequestToggleEnabled = featureToggleEnabled,
+                CreatedOn = request.CreatedOn,
+                Status = request.Status,
+                EnableStopRequestEmail = request.EnableStopRequestEmail
+            };
+            return View(vm);
+        }
+
+        [HttpPost]
+        [Route("overlap-options-with-pending-request")]
+        public async Task<IActionResult> DraftApprenticeshipOverlapOptionsWithPendingRequest(DraftApprenticeshipOverlapOptionWithPendingRequestViewModel viewModel)
+        {
+            var mappedViewModel = await _modelMapper.Map<DraftApprenticeshipOverlapOptionViewModel>(viewModel);
+            return await DraftApprenticeshipOverlapOptions(mappedViewModel);
```

Comment:
I would avoid calling another controller action from here. If there is shared logic then extract it to a private method and call from both

Reply:
updated (for minimum code changes I just moved method body to private method)